app/controllers/auth/registrations_controller.rb

```diff
@@ -28,7 +28,7 @@ def after_inactive_sign_up_path_for(_resource)
   end
 
   def check_enabled_registrations
-    redirect_to root_path if Rails.configuration.x.single_user_mode || !Setting.open_registrations
+    redirect_to root_path if Rails.configuration.x.single_user_mode || !InstancePresenter.new.open_registrations
```

Comment:
Maybe wrap this up into a private controller method like `single_user_or_registrations_closed?` to reveal the intention of the check here?